lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -1186,4 +1188,23 @@ public function quoteStringLiteral($str)
 
         return parent::quoteStringLiteral($str);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultValueDeclarationSQL($field)
+    {
+        if (self::isSerialField($field)) {
+            return '';
+        }
+
+        return parent::getDefaultValueDeclarationSQL($field);
+    }
+
+    private static function isSerialField($field) : bool
```

Comment:
Regarding the type, we just don't have the type declaration on `getDefaultValueDeclarationSQL` for historical reasons. Let's add it here too